Src/VimCore/EditorUtil.fs

```diff
@@ -43,6 +43,16 @@ module internal EditorCoreUtil =
     let IsInsideLineBreak (point: SnapshotPoint) (line: ITextSnapshotLine) = 
         point.Position >= line.End.Position && not (IsEndPoint point)
 
+    let EndsWithLineBreak (snapshot: ITextSnapshot) = 
+        let lineNumber = snapshot.LineCount - 1 
+        lineNumber > 0 && snapshot.GetLineFromLineNumber(lineNumber).Length = 0
+
+    let GetNormalizedLineCount (snapshot: ITextSnapshot) =

```

Comment:
Seems most of the functionality comes down to this line. May want to add a comment here explaining what the normalized line count is and why it's important.

Reply:
Good idea.